### PR TITLE
Add a new dust emission scheme from Kok et al. (2014)

### PR DESCRIPTION
**Pull request by yfenganl**
Add the new dust emission scheme from Kok et al., 2014.
The new scheme includes time-varying soil erodibility factor and high-latitude sources.
A namelist option is added to turn on the new emission scheme.

[stealth]
[BFB]
[NML] new shr_dust_nl section introduced.

### CODE REVIEW

**Comment by wlin7** [comment]
@mahf708 @rljacob @singhbalwinder , can you take another look of this PR, and approve it if you think ready?

BTW, [the simulations with this PR feature on look good to the atm evaluation team.](https://acme-climate.atlassian.net/wiki/spaces/NGDAP/pages/3712352357/Dust+emission)

**Comment by rljacob** [comment]
So there's namelist to turn on the feature but is it on by default?  If not, and there are no other effects from this PR, then this is BFB, Stealth.  Not CC.

**Comment by yfenganl** [comment]
> So there's namelist to turn on the feature but is it on by default? If not, and there are no other effects from this PR, then this is BFB, Stealth. Not CC.

Yes, that is correct. I will remove the CC labeling then

File: components/elm/src/biogeochem/DUSTMod.F90 [hunk 1]
```diff
@@ -241,9 +249,21 @@ subroutine DustEmission (bounds, &
...
+    real(r8), parameter :: Cd0 = 4.4e-5_r8             ! [dimless] proportionality constant 
+                                                       ! in calculation of dust emission coefficient -YF
+    real(r8), parameter :: Ca = 2.7_r8                 ! [dimless] proportionality constant in scaling of dust
+                                                       ! emission exponent -YF
+    real(r8), parameter :: Ce = 2.0_r8                 ! [dimless] proportionality constant scaling 
+                                                       ! exponential dependence of dust emission coefficient 
+                                                       ! on standardized soil threshold friction speed -YF
+    real(r8), parameter :: C_tune = 0.05_r8            ! [dimless] global tuning constant for vertical dust flux;
+                                                       ! set to produce ~same global dust flux in
+                                                       ! control sim (I_2000) as old parameterization -YF
```
[the remark this answers is not in the document]
Comment (yfenganl) [reply components/elm/src/biogeochem/DUSTMod.F90:264]: These parameters are calibrated together against observations, and I would prefer not to treat them as 'tunable' variables in the namelist.

[the remark this answers is not in the document]
Comment (rljacob) [reply components/elm/src/biogeochem/DUSTMod.F90:264]: You should add a comment indicating that those parameters need to be tuned together.

[the remark this answers is not in the document]
Comment (yfenganl) [reply components/elm/src/biogeochem/DUSTMod.F90:264]: in the namelist? or in the source file? I am not sure we know how to tune them, unless we have new observations.

[the remark this answers is not in the document]
Comment (rljacob) [reply components/elm/src/biogeochem/DUSTMod.F90:264]: In the source file as a note to future users.

[the remark this answers is not in the document]
Comment (yfenganl) [reply components/elm/src/biogeochem/DUSTMod.F90:265]: added a comment in the source file: "The following tuning constants are from Kok et al. (2014). They are optimized together."

File: components/elm/src/biogeochem/DUSTMod.F90 [hunk 2]
```diff
@@ -199,7 +202,8 @@ subroutine DustEmission (bounds, &
     ! from the surface into the lowest atmospheric layer
     ! On output flx_mss_vrt_dst(ndst) is the surface dust emission
     ! (kg/m**2/s) [ + = to atm]
-    ! Source: C. Zender's dust model
+    ! Original source: C. Zender's dust model
+    ! New emission scheme from Kok et al. (2014) implemented by Y. Feng in Feb, 2023
```
[the remark this answers is not in the document]
Comment (wlin7) [reply components/elm/src/biogeochem/DUSTMod.F90:206]: @yfenganl , would you like to add the DOI as @mahf708 suggested?

File: driver-mct/cime_config/namelist_definition_drv.xml [hunk 3]
```diff
@@ -5107,4 +5107,18 @@
...
+  <!-- ========================================================================================  -->
+
+  <entry id="dust_emis_scheme">
+    <type>integer</type>
+    <category>shr_dust</category>
+    <group>shr_dust_nl</group>
+    <desc>
+      Dust emission scheme.
+      1 = Zender et al. (2003) as in E3SMv1 and v2.
+    </desc>
```
[the remark this answers is not in the document]
Comment (wlin7) [reply driver-mct/cime_config/namelist_definition_drv.xml:5121]: I don't recall there is a policy for this.  In the codes, there are mixed uses of integer flag or string name. In theory, comparison of integer should go faster and that may help a little if the comparison is inside a loop.

File: components/elm/src/biogeochem/DUSTMod.F90 (hunk 2 above)
[the remark this answers is not in the document]
Comment (yfenganl) [reply components/elm/src/biogeochem/DUSTMod.F90:206]: I didn't see DOIs being added to the code previously. but if that's the policy now, sure, I am adding it.

File: components/elm/src/biogeochem/DUSTMod.F90 [hunk 4]
```diff
@@ -93,6 +94,8 @@ subroutine Init(this, bounds)
     call this%InitCold     (bounds)
     call this%InitDustVars (bounds)
 
+    write(iulog,*) "elm, DUSTMod.F90, Init: dust_emis_scheme = ",dust_emis_scheme
+
```
Comment (wlin7) [inline components/elm/src/biogeochem/DUSTMod.F90:98]: Same as above for eam's dust_model.F90 during init. Maybe useful to have both as you currently do, for the interest of running I-case.

**Review by wlin7 — APPROVED** [review]
The changes look good. Smoke tests for several different physics config went well, with the expected NLDIFF only, in a testing merge to next.

File: components/eam/src/chemistry/modal_aero/dust_model.F90 [hunk 5]
```diff
@@ -109,6 +111,8 @@ subroutine dust_init()
 
     call dust_set_params( dust_nbin, dust_dmt_grd, dust_dmt_vwr, dust_stk_crc )
 
+    write(iulog,*) "modal_aero, dust_init: dust_emis_scheme = ",dust_emis_scheme
```
[the remark this answers is not in the document]
Comment (yfenganl) [reply components/eam/src/chemistry/modal_aero/dust_model.F90]: @wlin7 Wuyin, I have made the suggested changes in eam

File: components/elm/src/biogeochem/DUSTMod.F90 (hunk 4 above)
Comment (yfenganl) [reply components/elm/src/biogeochem/DUSTMod.F90:99]: Also made the same change in elm.

**Comment by wlin7** [comment]
Merged to next.

**Comment by wlin7** [comment]
Re-merged to next after [83a5bc3](https://github.com/E3SM-Project/E3SM/pull/5531/commits/83a5bc3f7961d50719a8b3cde49ffb294234cff9)

**Comment by wlin7** [comment]
Merged to master.

**Comment by yfenganl** [comment]
Thank you Wuyin!

**Pull request merged**